spec/controllers/admin/videos_controller_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+
+  include CarrierWave::Test::Matchers
+
+  describe 'get :new' do
+
+    before(:each) do
+      admin = Fabricate(:admin)
+      login_user admin
+    end
+
+    it_behaves_like 'requires sign in' do
+      let(:action) { get :new }
+    end
+
+    it_behaves_like 'requires admin' do
+      let(:action) { get :new }
+    end
+
+    it 'sets @video variable' do
+      get :new
+      expect(assigns(:video)).to be_instance_of(Video)
+      expect(assigns(:video)).to be_new_record
+    end
+
+  end
+
+  describe 'post :create' do
+
+    it_behaves_like 'requires sign in' do
+      let(:action) { post :create }
+    end
+
+    it_behaves_like 'requires admin' do
+      let(:action) { post :create, video: {} }
+    end
+
+    context 'for valid values' do
+
+      let(:category) { Fabricate(:category) }
+      before(:each) do
+        admin = Fabricate(:admin)
+        login_user admin
+      end
+
+      context 'without images' do
+
+        let(:video_attrs) { Fabricate.attributes_for(:video, category_id: category.id) }
+        before do
+          post :create, video: video_attrs
+        end
+
+        it 'creates a video' do
+          expect(Video.all.count).to be 1
+        end
+
+        it 'associates the video with the passed category' do
+          expect(Video.last.category).to eq category
+        end
+
+        it 'redirects to the new video path' do
+          expect(response).to redirect_to new_admin_video_path
+        end
+
+        it 'sets the success message' do
+          expect(flash[:success]).to be
+        end
+      end
+
+      context 'with images' do
+
+        let(:video_attrs) { Fabricate.attributes_for(:video_with_covers_provided, category_id: category.id) }
+        before do
+          LargeCoverUploader.enable_processing = true
```

Comment:
because in the assignment video it's recommended to disable it for the test environment. I wanted to have control about it and enable it manually when I want to test uploaders' behavior.